### PR TITLE
Torch transformers support

### LINKED ISSUE #1303: [FEAT] Support torch transforms in the RobustnessReport
All the relevant code is under `deepchecks.vision.utils.transformations`

- [ ] We have an implementation of 2 classes that handles the transformations for albumentations and imgaug. Need to add a new class to handle torch transforms
- [ ] Add to `get_transforms_handler` the new class 
- [ ] Add test for robustness report with Dataset that works with torch transformations

### PR DESCRIPTION
**Pull request by JKL98ISR**
closes #1303

### CODE REVIEW

File: deepchecks/vision/datasets/classification/mnist.py [hunk 1]
```diff
@@ -44,7 +45,8 @@ def load_dataset(
     batch_size: t.Optional[int] = None,
     shuffle: bool = True,
     pin_memory: bool = True,
-    object_type: Literal['VisionData', 'DataLoader'] = 'DataLoader'
+    object_type: Literal['VisionData', 'DataLoader'] = 'DataLoader',
+    is_torch_transform: bool = False,
```
Comment (matanper) [inline deepchecks/vision/datasets/classification/mnist.py:49]: don't do this like this. Look how we did it in COCO

File: deepchecks/vision/utils/transformations.py [hunk 2]
```diff
@@ -118,24 +144,50 @@ def get_robustness_augmentations(cls, data_dim: Optional[int] = 3):
...
+class TorchTransformations(AlbumentationsTransformations):
+    """Class containing supporting functions for torch transforms."""
+
+    @classmethod
+    def add_augmentation_in_start(cls, aug, transforms: T.Compose):
+        """Add given transformations to the start of given transforms object."""
+        if isinstance(aug, (albumentations.Compose, albumentations.BasicTransform)):
+            alb_aug = aug
+
+            class TorchWrapper:
```
Comment (matanper) [inline deepchecks/vision/utils/transformations.py:156]: how this handle transformations on the label?

Comment (noamzbr) [reply deepchecks/vision/utils/transformations.py:156]: torch.transforms does not transform labels to the best of my knowledge, so makes sense we won't have this option when using an albumentation aug inside a torch pipeline...

File: deepchecks/vision/utils/transformations.py [hunk 3]
```diff
@@ -118,24 +144,50 @@ def get_robustness_augmentations(cls, data_dim: Optional[int] = 3):
         return augmentations
 
 
-def get_transforms_handler(transforms):
+class TorchTransformations(AlbumentationsTransformations):
```
Comment (noamzbr) [inline deepchecks/vision/utils/transformations.py:147]: TorchTransformations should have a different get_robustness_augmentations than AlbumentationsTransformations. 
Ideally, if it's classification we want the same things we had in AlbumentationsTransformations, and if it's any other task we want them excluding ShiftScaleRotate. Can we do that?

Comment (noamzbr) [reply deepchecks/vision/utils/transformations.py:147]: Also, why not use native torch augmentations for TorchTransformations?

File: tests/vision/checks/performance/robustness_report_test.py [hunk 4]
```diff
@@ -48,6 +49,49 @@ def test_mnist(mnist_dataset_train, mock_trained_mnist, device):
     }))
 
 
+def test_mnist_torch(mnist_dataset_train_torch, mock_trained_mnist, device):
```
Comment (noamzbr) [inline tests/vision/checks/performance/robustness_report_test.py:52]: We should also have a test for some RGB dataset.
For all I care, can create a fixture that has an version of MNIST that is RGB instead of greyscale.

Comment (matanper) [reply tests/vision/checks/performance/robustness_report_test.py:52]: We can just test on COCO

**Review by noamzbr — APPROVED** [review]

File: deepchecks/vision/utils/transformations.py [hunk 5]
```diff
@@ -118,24 +147,59 @@ def get_robustness_augmentations(cls, data_dim: Optional[int] = 3):
...
+                        return T.ToTensor()(alb_aug(image=image)['image'])
+                    return alb_aug(image=image)['image']
+            aug = TorchWrapper()
+        elif not isinstance(aug, torch.nn.Module):
+            raise DeepchecksValueError(f'Transforms is of type torch, can\'t add to it type '
+                                       f'{type(aug).__name__}')
+        return T.Compose([aug, *transforms.transforms])
+
+
+class TorchTransformations(TorchTransformationsClassification):
```
Comment (matanper) [inline deepchecks/vision/utils/transformations.py:174]: 1. Instead of having a different class maybe add task_type to `get_robustness_augmentations`
2. If you really want 2 classes, makes more sense that `TorchTransformationsClassification` will inherit from `TorchTransformations` and not the opposite

File: deepchecks/vision/utils/transformations.py [hunk 6]
```diff
@@ -9,21 +9,50 @@
...
-__all__ = ['get_transforms_handler', 'add_augmentation_in_start', 'un_normalize_batch',
+__all__ = ['get_transforms_handler', 'un_normalize_batch', 'AbstractTransformations',
            'ImgaugTransformations', 'AlbumentationsTransformations']
 
 
-class ImgaugTransformations:
+class AbstractTransformations(abc.ABC):
+    """Abstract class for supporting functions for various transforms."""
+
+    is_image_shifting = True
```
Comment (matanper) [inline deepchecks/vision/utils/transformations.py:34]: @noamzbr is "shifting" the only transformation that affects labels? I wonder if this should be called "is_transforming_labels" or something

Comment (noamzbr) [reply deepchecks/vision/utils/transformations.py:34]: yeah agree is_transforming_labels is better

File: deepchecks/vision/utils/transformations.py (hunk 5 above)
Comment (JKL98ISR) [reply deepchecks/vision/utils/transformations.py:174]: I want it to inherit from classification because then I can filter the transforms

Comment (matanper) [reply deepchecks/vision/utils/transformations.py:174]: so inherit both from Albumentations. this is too confusing IMO it's like `VisionData` will inherit from `DetectionData`

Comment (JKL98ISR) [reply deepchecks/vision/utils/transformations.py:174]: but both using same add_aug_at_start function

Comment (JKL98ISR) [reply deepchecks/vision/utils/transformations.py:174]: and I don't want to add task type as parameter, it make the usage more wonky and I need it in the class

File: deepchecks/vision/utils/transformations.py [hunk 7]
```diff
@@ -118,24 +147,71 @@ def get_robustness_augmentations(cls, data_dim: Optional[int] = 3):
...
+                def __call__(self, image):
+                    if isinstance(image, Image.Image):
+                        return T.ToPILImage()(alb_aug(image=np.array(image))['image'])
+                    elif isinstance(image, torch.Tensor):
+                        image = image.cpu().detach().numpy()
+                        return T.ToTensor()(alb_aug(image=image)['image'])
+                    return alb_aug(image=image)['image']
+            aug = TorchWrapper()
+        elif not isinstance(aug, torch.nn.Module):
+            raise DeepchecksValueError(f'Transforms is of type torch, can\'t add to it type '
```
Comment (matanper) [inline deepchecks/vision/utils/transformations.py:169]: this is a bit weird here, no? can't add torch transform inside TorchTransformations

**Review by matanper — APPROVED** [review]

File: deepchecks/vision/utils/transformations.py (hunk 7 above)
Comment (JKL98ISR) [reply deepchecks/vision/utils/transformations.py:169]: it's if not
